### PR TITLE
Add login page

### PR DESCRIPTION
**Pull request by subinasr**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -21,7 +21,12 @@
         <title>
             %APP_TITLE%
         </title>
-        <link href="https://api.fontshare.com/v2/css?f[]=nunito@300,700,400&f[]=bebas-neue@400&display=swap" rel="stylesheet">
+        <link rel="preconnect" href="https://fonts.googleapis.com">
+        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+        <link
+            href="https://fonts.googleapis.com/css2?family=Source+Sans+3:wght@300;400;700;900;&display=swap"
+            rel="stylesheet"
+        >
```
Comment (AdityaKhatri) [inline index.html:29]: add trailing slack if necessary

File: src/App/routes.tsx [hunk 2]
```diff
@@ -40,22 +40,61 @@ const home = myWrapRoute({
     title: 'Home',
     index: true,
     component: () => import('#views/Home'),
-    componentProps: { name: 'Pine Apple' },
+    componentProps: {},
+});
+
+const login = myWrapRoute({
+    title: 'Login',
+    path: 'login',
```
Comment (AdityaKhatri) [inline src/App/routes.tsx:65]:
```suggestion
    path: '/login',
```

File: src/components/FullPageErrorMessage/index.module.css [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    .logo-container {
+        display: flex;
+        align-items: center;
+        flex-direction: column;
+        z-index: 1;
+        max-width: 30rem;
+        text-align: center;
+
+        .logo {
+            width: 450px;
```
Comment (AdityaKhatri) [inline src/components/FullPageErrorMessage/index.module.css]: define in rem

File: src/components/HCaptcha/index.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            iconsContainerClassName={iconsContainerClassName}
+            inputSectionClassName={inputSectionClassName}
+            label={label}
+            labelContainerClassName={labelContainerClassName}
+            readOnly={readOnly}
+            uiMode={uiMode}
+            input={siteKey && (
+                <HCaptcha
+                    ref={elementRef}
+                    // disabled={disabled || readOnly}
```
Comment (AdityaKhatri) [inline src/components/HCaptcha/index.tsx:81]: Double-check if disabled is required or not.

File: src/configs/hCaptcha.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import {
+    hCaptchaKey as hCaptchaKeyFromEnv,
+} from './env.ts';
+
+// eslint-disable-next-line import/prefer-default-export
+export const hCaptchaKey = hCaptchaKeyFromEnv;
```
Comment (AdityaKhatri) [inline src/configs/hCaptcha.ts:6]: Lets remove this file and direcly use the key in hCaptcha file

File: src/redirects/ResetPasswordRedirect.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        state: {},
+    });
+
+    if (isNotDefined(resetPasswordLink)) {
+        return null;
+    }
+
+    return (
+        <Navigate
+            to="/reset-password"
```
Comment (AdityaKhatri) [inline src/redirects/ResetPasswordRedirect.tsx]: to should be used from wrappedRoutes.resetPassword as well.
Also, if pathName is not used in state, remove it and rename the link.

File: src/views/Register/index.tsx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+import { hCaptchaKey } from '#configs/hCaptcha';
+import {
+    RegisterMutation,
+    RegisterMutationVariables,
+    RegisterInput,
+} from '#generated/types';
+
+import styles from './index.module.css';
+
+// FIXME: import is weird will full names
```
Comment (AdityaKhatri) [inline src/views/Register/index.tsx]: Which import?

File: src/views/ResetPassword/index.tsx [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+        RESET_PASSWORD_CONFIRM,
+        {
+            onCompleted: (resetResponse) => {
+                const response = resetResponse?.public?.passwordResetConfirm;
+                if (!response) {
+                    return;
+                }
+                if (response.ok) {
+                    alert.show(
+                        'Logged in successfully!',
```
Comment (AdityaKhatri) [inline src/views/ResetPassword/index.tsx]:
```suggestion
                        'Password reset successfully!',
```

File: src/views/ResetPassword/index.tsx [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+                    siteKey={hCaptchaKey}
+                    // value={value.captcha}
+                    onChange={setFieldValue}
+                />
+                <Button
+                    name={undefined}
+                    type="submit"
+                    disabled={pristine || resetPasswordPending}
+                    onClick={handleSubmit}
+                    // FIXME: disable button when submitting
```
Comment (AdityaKhatri) [inline src/views/ResetPassword/index.tsx]: I think this is already done

File: src/views/ResetPassword/index.tsx [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+                    value={value.confirmPassword}
+                    error={safeError?.confirmPassword}
+                    onChange={setFieldValue}
+                    placeholder="Confirm password"
+                />
+                <HCaptcha
+                    name="captcha"
+                    elementRef={elementRef}
+                    siteKey={hCaptchaKey}
+                    // value={value.captcha}
```
Comment (AdityaKhatri) [inline src/views/ResetPassword/index.tsx:258]: This should be present

File: src/views/ResetPassword/index.tsx [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+            onCompleted: (resetResponse) => {
+                const response = resetResponse?.public?.passwordResetConfirm;
+                if (!response) {
+                    return;
+                }
+                if (response.ok) {
+                    alert.show(
+                        'Logged in successfully!',
+                        { variant: 'success' },
+                    );
```
Comment (AdityaKhatri) [inline src/views/ResetPassword/index.tsx:161]: We need to redirect the user to login page after password is reset.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (subinasr) [reply index.html:29]: Do you mean trailing slash? If yes, it's not required here.

File: src/App/routes.tsx (hunk 2 above)
Comment (subinasr) [reply src/App/routes.tsx:65]: Not required here, either.

File: src/App/routes.tsx [hunk 12]
```diff
@@ -3,59 +3,143 @@ import {
...
+        render: () => import('#views/ResetPassword'),
+        props: {},
+    },
+    context: {
+        title: 'Reset Password',
+        visibility: 'anything',
+    },
+});
+
+// FIXME: eager load this page
```
Comment (AdityaKhatri) [inline src/App/routes.tsx:104]: remove fixme

File: src/redirects/ResetPasswordRedirect.tsx [hunk 13]
```diff
@@ -0,0 +1,43 @@
+import { isNotDefined } from '@togglecorp/fujs';
+import {
+    Navigate,
+    generatePath,
+    useParams,
+} from 'react-router-dom';
+
+// import { wrappedRoutes } from '../App/routes.tsx';
```
Comment (AdityaKhatri) [inline src/redirects/ResetPasswordRedirect.tsx]: remove if not used

File: src/utils/localStorage.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
+export function getFromStorage<T>(key: string) {
```
Comment (AdityaKhatri) [inline src/utils/localStorage.ts]: I don't this is required any longer.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**